addons/sale_timesheet/tests/test_sale_timesheet.py

```diff
@@ -726,6 +726,72 @@ def test_unlink_timesheet(self):
         move.with_context(check_move_validity=False).line_ids[0].unlink()
         self.assertFalse(analytic_line.timesheet_invoice_id, "The timesheet should have been unlinked from move")
 
+    def test_sale_order_with_multiple_project_templates(self):
+        """Test when creating multiple projects for one sale order every project has its own allocated hours"""
+        sale_order = self.env['sale.order'].with_context(tracking_disable=True).create({
+            'partner_id': self.partner_a.id,
+            'partner_invoice_id': self.partner_a.id,
+            'partner_shipping_id': self.partner_a.id,
+        })
+
+        project_template_1, project_template_2 = self.env['project.project'].create([
+            {'name': 'Template 1'},
+            {'name': 'Template 1'},
+        ])
+
+        product_1, product_2 = self.env['product.product'].create([
+            {
+                'name': "Service with template 1",
+                'standard_price': 10,
+                'list_price': 20,
+                'type': 'service',
+                'invoice_policy': 'order',
+                'uom_id': self.uom_hour.id,
+                'uom_po_id': self.uom_hour.id,
+                'default_code': 'c1',
+                'service_tracking': 'task_in_project',
+                'project_id': False,  # will create a project,
+                'project_template_id': project_template_1.id,
+            }, {
+                'name': "Service with template 2",
+                'standard_price': 10,
+                'list_price': 20,
+                'type': 'service',
+                'invoice_policy': 'order',
+                'uom_id': self.uom_hour.id,
+                'uom_po_id': self.uom_hour.id,
+                'default_code': 'c2',
+                'service_tracking': 'task_in_project',
+                'project_id': False,  # will create a project,
+                'project_template_id': project_template_2.id,
+            },
+
```

Comment:
I don't know why you let a blank line here. 🤔 
It is not really needed.
```suggestion
            },
```